### PR TITLE
RFC: Substitute RocksDB write stall

### PR DESCRIPTION
**Pull request by Connor1996**
Substitute RocksDB write stall

[rendered](https://github.com/Connor1996/rfcs/blob/substitute-write-stall/text/2021-05-10-substitute-rocksdb-write-stall.md)

### CODE REVIEW

File: text/2021-05-10-substitute-rocksdb-write-stall.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Substitute rocksdb write stall
+
```
Comment (tisonkun) [inline text/2021-05-10-substitute-rocksdb-write-stall.md:2]:
```suggestion

- RFC PR: https://github.com/tikv/rfcs/pull/67
- Tracking Issue: https://github.com/tikv/tikv/issues/10137

```

**Comment by yiwu-arbug** [comment]
LGTM

**Review by yiwu-arbug — APPROVED** [review]

File: text/2021-05-10-substitute-rocksdb-write-stall.md (hunk 1 above)
Comment (tisonkun) [reply text/2021-05-10-substitute-rocksdb-write-stall.md:2]: I have done it for you as well as fixing the lint

PTAL https://github.com/Connor1996/rfcs/pull/1

Comment (Connor1996) [reply text/2021-05-10-substitute-rocksdb-write-stall.md:2]: Thanks!

**Review by zhangjinpeng1987 — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
@zhangjinpeng1987 you can merge the pull request if it is good to go. I think only a very small group of people has that right by now. To be precise, those who in [the maintainers team](https://github.com/orgs/tikv/teams/maintainers).

**Comment by mapleFU** [comment]
Hi, I wonder what will cause the memory high in tikv? 
And in "L0 files + memtables", seems it still has delay, why not discard the request here? @Connor1996

**Comment by mapleFU** [comment]
And:
> "For them, just check them periodically. IF it exceeds the threshold, then flow control is performed in AIMD(Additive Increase Multiplicative Decrease) way like TCP congestion control."

So the flow control is handled by TiDB? Does I catch it?

**Comment by Connor1996** [comment]
> And:
> 
> > "For them, just check them periodically. IF it exceeds the threshold, then flow control is performed in AIMD(Additive Increase Multiplicative Decrease) way like TCP congestion control."
> 
> So the flow control is handled by TiDB? Does I catch it?

Yes, TiDB will do retry.

**Pull request merged**